pkg/test/framework/components/istio/operator.go

```diff
@@ -707,19 +730,19 @@ func deployCACerts(workDir string, env *kube.Environment, cfg Config) error {
 			scopes.Framework.Infof("failed creating namespace %s on cluster %s. This can happen when deploying "+
 				"multiple control planes. Error: %v", cfg.SystemNamespace, cluster.Name(), err)
 		}
-
 		// Create the secret for the cacerts.
 		if _, err := cluster.CoreV1().Secrets(cfg.SystemNamespace).Create(context.TODO(), secret,
 			kubeApiMeta.CreateOptions{}); err != nil {
 			scopes.Framework.Infof("failed to create CA secrets on cluster %s. This can happen when deploying "+
 				"multiple control planes. Error: %v", cluster.Name(), err)
 		}
+
 	}
 	return nil
 }
 
-func configureDiscoveryForConfigCluster(discoveryAddress string, cfg Config, cluster resource.Cluster) error {
-	scopes.Framework.Infof("creating endpoints and service in %s to get discovery from %s", cluster.Name(), discoveryPort)
+func configureDiscoveryForConfigAndRemoteCluster(discoveryAddress string, cfg Config, cluster resource.Cluster) error {
```

Comment:
How can we make this more like what a user would do? I don't imagine we'll have them create endpoints/service by hand.

Reply:
added `"--set", "values.global.remotePilotAddress="+remoteIstiodAddress.IP.String()` to let it create the istiod endpoints automatically. Now this part is not used.

Reply:
Are we able to delete this method then?

Reply:
no.   As external Istiod test case is still relying on it.  ( for external Istiod case, we will install config cluster first and in that case it is not aware of the control plane istiod address, so "--set", "values.global.remotePilotAddress="+remoteIstiodAddress.IP.String()" will not work.)

Reply:
yep, so the challenge is we sometimes don't know istiod's endpoint ahead of time in our test framework.  If we do, we should not have this issue.  

I understand we have this issue for external istiod as it needs to set up config cluster first, but for MC primary-remote, you could setup primary first, then remote which will have the remoteIstiodAddress.   Is it possible not to use this path for MC primary-remote test?

Reply:
@irisdingbj `config cluster` != `remote cluster`. The installation of a config cluster should have no reference to istiod (i.e. does not use `values.global.remotePilotAddress`). We should safely be able to install clusters as follows:

1. Install config clusters (basic needed for external istiod)
1. Install primary clusters
1. Install remote clusters (may or may not be the same clusters as the config clusters)

I think we should change this here in this PR unless there is an overwhelming reason why it can't be done now.

Reply:
yes, now only remote cluster needs `values.global.remotePilotAddress`. Now for externalIstiod test case, we are using the same cluster for both config and remote cluster.  We can modify this later to use 3 clusters to distinguish between config and remote cluster.

Reply:
yes, now only remote cluster needs values.global.remotePilotAddress. Now for externalIstiod test case, we are using the same cluster for both config and remote cluster. We can modify this later to use 3 clusters to distinguish between config and remote cluster.

Reply:
Agree config cluster doesn't need remotePilotAddr.  Only discoverAddr

Reply:
It shouldn't need either `discoveryAddr` or `remotePilotAddr`. The config cluster is inherently the first thing that should be configured because the control plane relies on it (not the other way around). We have to fix this before 1.9.